tests/test_cms_menus.py

```diff
@@ -250,6 +253,115 @@ def test_get_roots_with_versioning_disabled(self):
             list(roots), [menucontent_1.root, menucontent_2.root, menucontent_3.root]
         )
 
+    def test_draft_menu_on_draft_page(self):
+        """
+        Draft menu changes on draft page rendering.
+        """
+        menu_cont_draft = factories.MenuContentWithVersionFactory(version__state=DRAFT, language=self.language)
+        menu_cont_published = factories.MenuContentWithVersionFactory(version__state=PUBLISHED, language=self.language)
+        pagecontent_aaa = factories.PageContentWithVersionFactory(
+            language=self.language,
+            version__created_by=self.get_superuser(),
+            title="aaa",
+            menu_title="aaa",
+            page_title="aaa",
+            version__state=DRAFT,
+        )
+        pagecontent_bbb = factories.PageContentWithVersionFactory(
```

Comment:
pagecontent_bbb made sense for the menu where they where logical but here it doesn't really make sense. Make it easy for the reader to follow things, pagecontent_published / draft helps here as you aren't changing states.